### PR TITLE
Do not substitute parameters during SQL/Expression formatting

### PR DESCRIPTION
**Pull request by martint**
This is the wrong place to do so. The formatter should do a literal
translation from the syntax tree to the textual representation.

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/execution/DataDefinitionTask.java [hunk 1]
```diff
@@ -33,11 +32,21 @@
...
-        if (statement instanceof Prepare) {
-            return SqlFormatter.formatSql(statement, Optional.empty());
+        StringBuilder builder = new StringBuilder();
+
+        builder.append(SqlFormatter.formatSql(statement));
+
+        if (!parameters.isEmpty()) {
+            String formatted = parameters.stream()
+                    .map(Expression::toString)
+                    .collect(Collectors.joining(", ", "[", "]"));
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/execution/DataDefinitionTask.java]: formatted eq parameters.toString()

(List#toString is documented, stable, AFAIK)

in fact, you don't use `formatted` at all

(also, static import .joining)

File: presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java [hunk 2]
```diff
@@ -270,8 +269,8 @@ public void parseUnknownRules()
                 .hasMessageContaining("Invalid JSON");
     }
 
-    private SystemAccessControl parse(String path)
+    private void parse(String path)
     {
-        return new FileBasedSystemAccessControl.Factory().create(ImmutableMap.of(SECURITY_CONFIG_FILE, path));
+        new FileBasedSystemAccessControl.Factory().create(ImmutableMap.of(SECURITY_CONFIG_FILE, path));
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java]: unrelated change

File: presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java [hunk 3]
```diff
@@ -104,7 +104,6 @@ public void testCanSetUserOperations()
         catch (AccessDeniedException expected) {
         }
 
-        TransactionManager transactionManagerNoPatterns = createTestTransactionManager();
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java]: unrelated change

**Review by findepi — APPROVED** [review]

File: presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java (hunk 3 above)
Comment (martint) [reply presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java]: Weird, no idea how these happened.

File: presto-main/src/main/java/io/prestosql/execution/DataDefinitionTask.java (hunk 1 above)
Comment (martint) [reply presto-main/src/main/java/io/prestosql/execution/DataDefinitionTask.java]: Left-over from some work-in-progress...
